src/main/java/io/github/thebusybiscuit/slimefun4/implementation/operations/MiningOperation.java

```diff
@@ -0,0 +1,45 @@
+package io.github.thebusybiscuit.slimefun4.implementation.operations;
+
+import javax.annotation.Nonnull;
+
+import org.apache.commons.lang.Validate;
+import org.bukkit.inventory.ItemStack;
+
+import io.github.thebusybiscuit.slimefun4.core.machines.MachineOperation;
+
+public class MiningOperation implements MachineOperation {
+
+    private final ItemStack result;
+
+    private final int totalTicks;
+    private int currentTicks = 0;
+
+    public MiningOperation(@Nonnull ItemStack result, int totalTicks) {
+        Validate.notNull(result, "The result cannot be null");
+        Validate.isTrue(totalTicks > 0, "The amount of total ticks must be a positive integer");
+
+        this.result = result;
+        this.totalTicks = totalTicks;
+    }
+
+    @Override
+    public void addProgress(int num) {
```

Comment:
Maybe a check to make sure its a positive number?
Or start using a uint I suppose